source/dub/compilers/dmd.d

```diff
@@ -156,23 +156,23 @@ class DmdCompiler : Compiler {
 
 	void setTarget(ref BuildSettings settings, in BuildPlatform platform, string tpath = null) const
 	{
-		final switch (settings.targetType) {
-			case TargetType.autodetect: assert(false, "Invalid target type: autodetect");
-			case TargetType.none: assert(false, "Invalid target type: none");
-			case TargetType.sourceLibrary: assert(false, "Invalid target type: sourceLibrary");
-			case TargetType.executable: break;
-			case TargetType.library:
-			case TargetType.staticLibrary:
-				settings.addDFlags("-lib");
-				break;
-			case TargetType.dynamicLibrary:
-				version (Windows) settings.addDFlags("-shared");
-				else settings.addDFlags("-shared", "-fPIC");
-				break;
-		}
-
-		if (tpath is null)
+		if (tpath is null) {
```

Comment:
I'm not sure I understand the logic here.
There are 2 very different codepath for this function, based on `tpath`. Looks like this function is being used for 2 differents purpose, isn't it ?